tests/integration/targets/attachment/tasks/main.yml

```diff
@@ -0,0 +1,470 @@
+- environment:
+    SN_HOST: "{{ sn_host }}"
+    SN_USERNAME: "{{ sn_username }}"
+    SN_PASSWORD: "{{ sn_password }}"
+
+  block:
+    #############################################
+    ###### Change Request Attachment Tests ######
+    #############################################
+
+    - name: Create test change_request with attachment - check mode
+      servicenow.itsm.change_request: &create-attachment
+        requested_by: admin
+        state: new
+        type: standard
+        template: Clear BGP sessions on a Cisco router - 1
+        priority: low
+        risk: low
+        impact: low
+        attachments:
+          - path: targets/attachment/res/sample_file.txt
+      check_mode: true
+      register: result
+
+    - ansible.builtin.assert: &create-attachment-result
+        that:
+          - result is changed
+          - result.record.state == "new"
+          - result.record.type == "standard"
+          - result.record.priority == "low"
+          - result.record.risk == "low"
+          - result.record.impact == "low"
+          - result.record.attachments != []
+          - result.record.attachments | length == 1
+          - result.record.attachments[0].file_name == "sample_file.txt"
+
+
+    - name: Create test change_request with attachment
+      servicenow.itsm.change_request: *create-attachment
+      register: test_result
+
+    - name: Copy test_result into result for assertion
+      set_fact:
+        result: "{{ test_result }}"
+
+    - ansible.builtin.assert: *create-attachment-result
+
+
+    - name: Update change_request with same attachment, same name, add new (check_mode)
+      servicenow.itsm.change_request: &cr-update
+        number: "{{ test_result.record.number }}"
+        attachments:
+          - path: targets/attachment/res/sample_file.txt
+            name: sample_file.txt
+          - path: targets/attachment/res/sample_file.txt
+            name: sample_file2.txt
+      check_mode: true
+      register: result
+
+    - ansible.builtin.assert:
+        that:
+          - result is changed
+          - result.record.attachments[0].file_name == "sample_file.txt"
+          - result.record.attachments[1].file_name == "sample_file2.txt"
```

Comment:
Order can be problematic here. We do not guarantee a specific order of attachments.

```suggestion
          - result.record.attachments | map(attribute="file_name") | sort | list == ["sample_file.txt", "sample_file2.txt"]
```

Also, maybe we should use different test file names because I must admit I am not sure if I sorted the current ones correctly. Something like sample_01, sample_02, ..., for old people like me ;)

This ordering problem problem holds for all tests.